type-check.ts

```diff
@@ -885,6 +889,25 @@ export function tcExpr(
               methodArgs.every((argTyp, i) => isAssignable(env, realArgs[i].a[0], argTyp))
             ) {
               return { ...expr, a: [methodRet, expr.a], obj: tObj, arguments: tArgs };
+            } else if (
```

Comment:
I believe this is similar to how you would handle defaults with the `call` tag. This feels like duplicated code.

Reply:
Hey Derek, thanks for your comment. I will look over it and try to move this logic to a function